File: spec/javascript/test_imports_circular_a.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('ImportsCircularA', 'src/fixed_struct.bin', function(r) {
  assert.equal(r.code, 0x50);
  assert.equal(r.two.initial, 0x41);
  assert.equal(r.two.backRef.code, 0x43);
  assert.equal(r.two.backRef.two.initial, 0x4b);
  assert.strictEqual(r.two.backRef.two.backRef, null);
});

File: spec/python/test_imports_circular_a.py
import unittest

from imports_circular_a import ImportsCircularA

class TestImportsCircularA(unittest.TestCase):
    def test_imports_circular_a(self):
        r = ImportsCircularA.from_file("src/fixed_struct.bin")

        self.assertEqual(r.code, 0x50)
        self.assertEqual(r.two.initial, 0x41)
        self.assertEqual(r.two.back_ref.code, 0x43)
        self.assertEqual(r.two.back_ref.two.initial, 0x4b)
        self.assertFalse(hasattr(r.two.back_ref.two, 'back_ref'))
